### PR TITLE
docs: sd-brandshape docs [skip chromatic]

### PR DESCRIPTION
**Pull request by smfonseca**
## Definition of Reviewable:

- [ ] Documentation is created/updated
- [ ] Migration Guide is created/updated
- [ ] E2E tests (features, a11y, bug fixes) are created/updated
- [ ] Stories (features, a11y) are created/updated
- [ ] relevant tickets are linked

### CODE REVIEW

**Comment by mariohamann** [comment]
Ohoh. I think we have to merge `main` into the other branch. There are two other variants:

https://solid-design-system.fe.union-investment.de/x.x.x/storybook/?path=/docs/components-sd-brandshape--docs#variants

File: packages/components/src/components/brandshape/brandshape.stories.ts [hunk 1]
```diff
@@ -1,15 +1,22 @@
...
 import { storybookDefaults, storybookHelpers, storybookTemplate } from '../../../scripts/storybook/helper';
-import type { ConstantDefinition } from '../../../scripts/storybook/helper';
 
 const { argTypes, parameters } = storybookDefaults('sd-brandshape');
 const { overrideArgs } = storybookHelpers('sd-brandshape');
 const { generateTemplate } = storybookTemplate('sd-brandshape');
 
+/**
+ *
+ * This component is used to create a brand box with a background and create a visual separation in the UI.
```
Comment (coraliefeil) [inline packages/components/src/components/brandshape/brandshape.stories.ts:13]: Creates a container with brand character and visually emphasizes the content.

File: packages/components/src/components/brandshape/brandshape.stories.ts [hunk 2]
```diff
@@ -30,125 +37,78 @@ export default {
...
+        <p class="sd-headline sd-headline--lg">White Variant Example</p>
+        <p>Example content text</p>
+      </sd-brandshape>
+    </div>
+  `
 };
 
 /**
- * Use the `base`, `content`, `shape-top`, `shape-middle` or `shape-bottom` part selectors to customize the brandshape.
+ * The different shapes of the brandshape. By default, all shapes are displayed but it's possible to display only specific shapes by utilizing the `shapes` attribute.
```
Comment (coraliefeil) [inline packages/components/src/components/brandshape/brandshape.stories.ts]: Use the ´´shapes´´ attribute to only display specific parts of the brandshape.

File: packages/components/src/components/brandshape/brandshape.stories.ts [hunk 3]
```diff
@@ -30,125 +42,90 @@ export default {
...
+        <p class="sd-headline sd-headline--lg">Image Variant Example</p>
+        <img slot="image" style="transform:translateY(-30%);" src="./placeholders/images/generic.jpg" alt="Generic" />
+      </sd-brandshape>
+    </div>
+  `
 };
 
 /**
- * Use the `base`, `content`, `shape-top`, `shape-middle` or `shape-bottom` part selectors to customize the brandshape.
+ * By default, all shapes are displayed but it's possible to display only specific shapes by utilizing the `shapes` attribute.
```
Comment (coraliefeil) [inline packages/components/src/components/brandshape/brandshape.stories.ts:102]: Code-only: Use the “shapes” attribute to only display specific parts of the brandshape.

Figma-only: Hide layers to only display specific parts of the brandshape.

**Comment by coraliefeil** [comment]
See comment.
Design docs are done as well. Pls have a look.

File: packages/components/src/components/brandshape/brandshape.stories.ts [hunk 4]
```diff
@@ -35,157 +42,90 @@ export default {
...
-      },
-      args,
-      constants: increaseColumnWidth()
-    });
-  }
-};
+export const Variant = {
+  name: 'Variant',
+  render: () => html`
+    <div class="mb-5">
```
Comment (mariohamann) [inline packages/components/src/components/brandshape/brandshape.stories.ts:62]: Can you use one div for all elements with `space-y-5`?

File: packages/components/src/components/brandshape/brandshape.stories.ts [hunk 5]
```diff
@@ -35,157 +42,90 @@ export default {
...
+      <sd-brandshape variant="primary">
+        <p class="sd-headline sd-headline--lg text-white">Primary Variant Example</p>
```
Comment (mariohamann) [inline packages/components/src/components/brandshape/brandshape.stories.ts:64]: @coraliefeil @smfonseca What do you think about using the claim "We work for your investment" on all brand shapes, maybe even in bold and bigger? Or do we have other examples? Maybe we could even show, that the big brand shape would have more content and the others less?

CleanShot 2024-08-06 at 11 41 23@2x

File: packages/components/src/components/brandshape/brandshape.stories.ts (hunk 3 above)
Comment (smfonseca) [reply packages/components/src/components/brandshape/brandshape.stories.ts:102]: Updated

File: packages/components/src/components/brandshape/brandshape.stories.ts (hunk 4 above)
Comment (smfonseca) [reply packages/components/src/components/brandshape/brandshape.stories.ts]: Updated

**Review by mariohamann — APPROVED** [review]
If you're in sync and fine, I'm too. @smfonseca

**Comment by yoezlem** [comment]
Please align with design.

For Variants they start with: neutral-100, primary, primary|80 (?), border-primary. They also have more variants then we have in Storybook.

**Comment by yoezlem** [comment]
The slot in the "top" part of the sd-brandshape needs to be removed.
image

And I am not sure, but the "bottom" part makes some weird thing with the slot.

Only the "middle" section/part should have a slot

**Comment by coraliefeil** [comment]
> The slot in the "top" part of the sd-brandshape needs to be removed. image
> 
> And I am not sure, but the "bottom" part makes some weird thing with the slot.
> 
> Only the "middle" section/part should have a slot

Wait ... in the superteaser there is 1 singular case where there is the slot in the top shape .... we did not implement it in figma because the use case was not general enough ...

**Review by Vahid1919 — APPROVED** [review]

**Comment by coraliefeil** [comment]
I don’t know how it is built in code but[ in figma in the cms library](https://www.figma.com/design/QjYmW5zZuGLdnSibmDD2Fo/UI-CMS-Library?node-id=1692-10121&t=6KJ3TTnmKf4ck45m-11) they are not using sd-brandshape (because it does not offer the feature of slot in top shape)

**Comment by mariohamann** [comment]
> I don’t know how it is built in code but[ in figma in the cms library](https://www.figma.com/design/QjYmW5zZuGLdnSibmDD2Fo/UI-CMS-Library?node-id=1692-10121&t=6KJ3TTnmKf4ck45m-11) they are not using sd-brandshape (because it does not offer the feature of slot in top shape)

On dev side it's in use. This happened when @christophsaile implemented it on our side. I would say: Maybe we should just document it on design side, too – without implementing it, if it's too hard?

**Comment by yoezlem** [comment]
> Please align with design.
> 
> For Variants they start with: neutral-100, primary, primary|80 (?), border-primary. They also have more variants then we have in Storybook.
> 
> Figma: image
> 
> Storybook: image
> 
> They look different to me. Shouldn't the docs for Figma and Storybook be identical?

If we don't resolve this here, please create a ticket for it and assign it to me.

**Comment by MartaPintoTeixeira** [comment]
Regarding the Variants me and @coraliefeil  have refined our Figma doc today. 
Please have a look and follow the same order in CODE please.

**Comment by coraliefeil** [comment]
Approved. THX

File: packages/components/src/components/brandshape/brandshape.stories.ts [hunk 6]
```diff
@@ -1,15 +1,25 @@
...
 const { argTypes, parameters } = storybookDefaults('sd-brandshape');
 const { overrideArgs } = storybookHelpers('sd-brandshape');
 const { generateTemplate } = storybookTemplate('sd-brandshape');
 
+/**
+ *
+ * Creates a container with brand character and visually emphasizes the content.
+ *
+ * **Related templates:**
+ * - [Brandshape Template](?path=/story/templates-brandshape-template--docs)
```
Comment (yoezlem) [inline packages/components/src/components/brandshape/brandshape.stories.ts:16]:
```suggestion
 * - [Brandshape](?path=/story/templates-brandshape-template--docs)
```

**Review by yoezlem — COMMENTED** [review]
@smfonseca did you create a ticket for the transparent variants?
image

**Comment by yoezlem** [comment]
Rest LGTM

**Comment by coraliefeil** [comment]
I’m not sure if that really works in code:

The placeholder image is distorted:
Bildschirmfoto 2024-09-02 um 09 12 30

The question is: Why is there a default slot? That needs to be deleted in code IMO. Looking at the old Sketch Library there’s only the brandshape used to mask an image. Nothing else. So placing content on top of it is not intended. Also there is no overlay providing sufficient contrast between content and image ...

**Comment by coraliefeil** [comment]
Could we add the templates at at later point of time, pls? Is there already an issue handling it?
See [Figma](https://www.figma.com/design/YDktJcseQIIQbsuCpoKS4V/Component-Docs?node-id=3121-11350&node-type=SECTION&t=SwfvfYHM9JnxEX6r-11)

**Comment by coraliefeil** [comment]
[primary (default): used on light backgrounds -> st](https://solid-design-system.github.io/solid/feature_sd-brandshape-docs/?path=/docs/components-sd-brandshape--docs#variant)
Show primary first and underneath neutral-100

**Comment by smfonseca** [comment]
> @smfonseca did you create a ticket for the transparent variants? image

@yoezlem these variants do not exist in code, they need to be created

**Comment by smfonseca** [comment]
> Could we add the templates at at later point of time, pls? Is there already an issue handling it? See [Figma](https://www.figma.com/design/YDktJcseQIIQbsuCpoKS4V/Component-Docs?node-id=3121-11350&node-type=SECTION&t=SwfvfYHM9JnxEX6r-11)

@coraliefeil the first template is already created, for the second one we need to create the transparent variants. I will create an issue for it and include the reference to the template.

**Comment by smfonseca** [comment]
> I’m not sure if that really works in code:
> 
> The placeholder image is distorted: Bildschirmfoto 2024-09-02 um 09 12 30
> 
> The question is: Why is there a default slot? That needs to be deleted in code IMO. Looking at the old Sketch Library there’s only the brandshape used to mask an image. Nothing else. So placing content on top of it is not intended. Also there is no overlay providing sufficient contrast between content and image ...

As discussed in today's sync call, I will create an issue to remove the slot when the variant is an image.
FYI @yoezlem

**Review by yoezlem — APPROVED** [review]

File: packages/components/src/components/brandshape/brandshape.stories.ts [hunk 7]
```diff
@@ -1,15 +1,25 @@
...
 import { storybookDefaults, storybookHelpers, storybookTemplate } from '../../../scripts/storybook/helper';
-import type { ConstantDefinition } from '../../../scripts/storybook/helper';
 
 const { argTypes, parameters } = storybookDefaults('sd-brandshape');
 const { overrideArgs } = storybookHelpers('sd-brandshape');
 const { generateTemplate } = storybookTemplate('sd-brandshape');
 
+/**
+ *
+ * Creates a container with brand character and visually emphasizes the content.
```
Comment (MartaPintoTeixeira) [inline packages/components/src/components/brandshape/brandshape.stories.ts:13]: Used to create a container with brand character and visually emphasizes the content.

**Pull request merged**